cpp/src/Slice/Grammar.y

```diff
@@ -1264,13 +1272,22 @@ operation
 {
     if ($1)
     {
+        // Check that all out parameters come before all in parameters.
+        auto op = dynamic_pointer_cast<Operation>($1);
+        bool seenOutParam = false;
+        for (const auto& param : op->parameters())
+        {
+            const bool isOutParam = param->isOutParam();
+            if (!isOutParam && seenOutParam)
+            {
+                currentUnit->error("parameter '" + param->name() + "': in-parameters cannot follow out-parameters");
+            }
+            seenOutParam |= isOutParam;
+        }
+
```

Comment:
We used to this check in `Parser.cpp: createParameter`, called once for each parameter.
Now we only do it once. Right here, when we've finished creating an operation.

I think this is cleaner.
But was also necessary. One of the things I changed is that we don't set `isOutParam` until _after_ we've initialized a parameter now.